### PR TITLE
Dynamically Link Unit Tests to Speed Up Builds

### PR DESCRIPTION
**Pull request by chenhao94**
Warning: This will break some of the unit tests on Mac OS, but since we are prioritizing Linux so we will fix MacOS issues later.

Issue:
- https://github.com/cyfitech/cris-core/issues/239

### CODE REVIEW

File: bazel/rules.bzl [hunk 1]
```diff
@@ -38,4 +35,6 @@ def cris_cc_binary(**attrs):
     native.cc_binary(**__add_opts(attrs))
 
 def cris_cc_test(**attrs):
+    if "linkstatic" not in attrs:
```
Comment (xkszltl) [inline bazel/rules.bzl:38]: Can it exist but empty?

**Review by xkszltl — APPROVED** [review]

File: bazel/rules.bzl [hunk 2]
```diff
@@ -38,4 +35,6 @@ def cris_cc_binary(**attrs):
...
+        attrs["linkstatic"] = False
```
Comment (xkszltl) [inline bazel/rules.bzl:39]: - Do we need to specify anything on lib side? 
- Not every lib should be linked as shared, especially protobuf.

Comment (chenhao94) [reply bazel/rules.bzl:39]: > - Do we need to specify anything on lib side? 

not for now 

> - Not every lib should be linked as shared, especially protobuf.

That is exactly why not

File: bazel/rules.bzl (hunk 1 above)
Comment (chenhao94) [reply bazel/rules.bzl:38]: No, it must be a boolean value.

**Pull request merged**